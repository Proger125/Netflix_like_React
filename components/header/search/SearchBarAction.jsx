import React from 'react';
import { Formik, Form, Field } from 'formik';
import { useRouter } from 'next/router';
import PropTypes from 'prop-types';

export default function SearchBarAction(props) {
  const { onSubmit } = props;
  const router = useRouter();
  const { search, genre, sortBy } = router.query;
  const searchParam = search === undefined ? '' : search;
  return (
    <Formik
      initialValues={{
        search: searchParam,
      }}
      onSubmit={(values) => {
        onSubmit(values.search, sortBy, genre);
      }}
    >
      {({ values, handleChange }) => (
        <Form className="searchbar-action">
          <Field
            className="searchbar-input"
            type="text"
            name="search"
            placeholder="What do you want to watch?"
            value={values.search}
            onChange={handleChange}
          />
          <button className="searchbar-button" type="submit">
            Search
          </button>
        </Form>
      )}
    </Formik>
  );
}

SearchBarAction.propTypes = {
  onSubmit: PropTypes.func.isRequired,
};
