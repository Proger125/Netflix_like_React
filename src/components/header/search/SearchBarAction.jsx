import React from 'react';
import { Formik, Form, Field } from 'formik';
import { useLoaderData } from 'react-router-dom';
import PropTypes from 'prop-types';

export default function SearchBarAction(props) {
  const { onSubmit } = props;
  const { searchQuery, genreParam, sortBy } = useLoaderData();
  return (
    <Formik
      initialValues={{
        search: searchQuery,
      }}
      onSubmit={(values) => {
        const genre = genreParam;
        onSubmit(values.search, sortBy, genre);
      }}
    >
      {({ values, handleChange }) => (
        <Form className="searchbar-action">
          <Field
            className="searchbar-input"
            type="text"
            name="search"
            placeholder="What do you want to watch?"
            value={values.search}
            onChange={handleChange}
          />
          <button className="searchbar-button" type="submit">
            Search
          </button>
        </Form>
      )}
    </Formik>
  );
}

SearchBarAction.propTypes = {
  onSubmit: PropTypes.func.isRequired,
};
