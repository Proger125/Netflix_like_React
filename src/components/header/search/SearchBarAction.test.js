/* eslint-disable react/jsx-filename-extension */
/* eslint-disable no-undef */

import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import { createBrowserRouter, RouterProvider } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import SearchBarAction from './SearchBarAction';

describe('Search Bar actions tests', () => {
  let router;
  const mockesOnSubmit = jest.fn();
  beforeAll(() => {
    jest.mock('react-router-dom', () => ({
      ...jest.requireActual('react-router-dom'),
      useLoaderData: () => ({
        searchQuery: '',
        genreParam: '',
        sortBy: '',
      }),
      useNavigate: () => jest.fn(),
    }));
    jest.mock('formik', () => ({
      ...jest.requireActual('formik'),
    }));
    router = createBrowserRouter([
      {
        path: '/',
        element: <SearchBarAction onSubmit={mockesOnSubmit} />,
        loader: () => {
          return { searchQuery: '' };
        },
      },
    ]);
  });

  test('renders search bar actions component', () => {
    render(<RouterProvider router={router} />);
    expect(screen.getByText('Search')).toBeInTheDocument();
  });

  test('renders search bar actions component and submit search', async () => {
    render(<RouterProvider router={router} />);
    const input = screen.getByPlaceholderText('What do you want to watch?');
    const searchButton = screen.getByText('Search');
    await userEvent.type(input, 'Movie');
    await userEvent.click(searchButton);
    expect(input.value).toBe('Movie');
    await waitFor(() => {
      expect(mockesOnSubmit).toHaveBeenCalled();
    });
  });
});
