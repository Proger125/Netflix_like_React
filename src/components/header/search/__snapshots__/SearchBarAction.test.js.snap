// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Bar actions tests snapshot test 1`] = `
<form
  action="#"
  className="searchbar-action"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <input
    className="searchbar-input"
    name="search"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="What do you want to watch?"
    type="text"
    value=""
  />
  <button
    className="searchbar-button"
    type="submit"
  >
    Search
  </button>
</form>
`;
